lib/patches/bundler/ruby_version.rb

```diff
@@ -0,0 +1,29 @@
+module Bundler
+  class RubyVersion
+    def self.system
+      ruby_engine         = if defined?(RUBY_ENGINE) && !RUBY_ENGINE.nil?
+                              RUBY_ENGINE.dup
+                            else
+                              # not defined in ruby 1.8.7
+                              "ruby"
+                            end
+      # :sob: mocking RUBY_VERSION breaks stuff on 1.8.7
+      ruby_version        = ENV.fetch("BUNDLER_SPEC_RUBY_VERSION") { RUBY_VERSION }.dup
+      ruby_engine_version = case ruby_engine
+                            when "ruby"
+                              ruby_version
+                            when "rbx"
+                              Rubinius::VERSION.dup
+                            when "jruby"
+                              JRUBY_VERSION.dup
+                            when 'truffleruby' # added case branch
```

Comment:
It would be good to use a common prefix to mean "comment added for TruffleRuby support".
We used to have `# Truffle: my comment` which I think is easy to spot.
I think we should be consistent.